crates/storage/provider/src/traits/storage.rs

```diff
@@ -0,0 +1,33 @@
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    ops::RangeInclusive,
+};
+
+use auto_impl::auto_impl;
+use reth_interfaces::Result;
+use reth_primitives::{Address, BlockNumber, StorageEntry, H256};
+
+/// Storage reader
+#[auto_impl(&, Arc, Box)]
+pub trait StorageReader: Send + Sync {
+    /// Get plainstate storages from an Address.
```

Comment:
I think it should be plural and mentioning the slots, as we can retrieve multiple slots per multiple addresses
```suggestion
    /// Get plainstate storages for addresses and storage keys.
```